### DISCUSSION TITLE
Formatter and Text Cleaner

### DISCUSSION THREAD

**Discussion by koayst** [coqui-ai/TTS#1818] [General Q&A] [answered]
Hi,

From "Tutorial_2_train_your_first_TTS_model.ipynb", it is mentioned that:

"After you collect and format your dataset, you need to check two things. Whether you need a **formatter** and a **text_cleaner**.
The **formatter** loads the text file (created above) as a list and the **text_cleaner** performs a sequence of text normalization operations that converts the raw text into the spoken representation (e.g. converting numbers to text, acronyms, and symbols to the spoken format)."

I think I found "**text_cleaner**" - TTS/tts/utils/text/cleaners.py.

I am not sure where is "**formatter**".  Can someone point me to the right place/file please ?

Thanks

ST

**Comment by harmlessman** [comment, marked as answer]
The location of `formatters.py` is as follows

`TTS\\tts\\datasets\\formatters.py`

**Reply by koayst** [reply]
Thanks.
